backend/src/schema/resolvers/rooms.spec.ts

```diff
@@ -0,0 +1,198 @@
+import { createTestClient } from 'apollo-server-testing'
+import Factory, { cleanDatabase } from '../../db/factories'
+import { getNeode, getDriver } from '../../db/neo4j'
+import { createRoomMutation, roomQuery } from '../../graphql/rooms'
+import createServer from '../../server'
+
+const driver = getDriver()
+const neode = getNeode()
+
+let query
+let mutate
+let authenticatedUser
+let chattingUser, otherChattingUser, notChattingUser
+
+beforeAll(async () => {
+  await cleanDatabase()
+
+  const { server } = createServer({
+    context: () => {
+      return {
+        driver,
+        neode,
+        user: authenticatedUser,
+      }
+    },
+  })
+  query = createTestClient(server).query
+  mutate = createTestClient(server).mutate
+})
+
+afterAll(async () => {
+  // await cleanDatabase()
+  driver.close()
+})
+
+describe('Room', () => {
+  beforeAll(async () => {
+    ;[chattingUser, otherChattingUser, notChattingUser] = await Promise.all([
+      Factory.build(
+        'user',
+        {
+          id: 'chatting-user',
+          name: 'Chatting User',
+        },
+      ),
+      Factory.build(
+        'user',
+        {
+          id: 'other-chatting-user',
+          name: 'Other Chatting User',
+        },
+      ),
+      Factory.build(
+        'user',
+        {
+          id: 'not-chatting-user',
+          name: 'Not Chatting User',
+        },
+      ),
+    ])
+  })
+
+  describe('create room', () => {
+    describe('unauthenticated', () => {
+      it('throws authorization error', async () => {
+        await expect(mutate({ mutation: createRoomMutation(), variables: {
+          userId: 'some-id' } })).resolves.toMatchObject({
+            errors: [{ message: 'Not Authorized!' }],
+          })
+      })
+    })
+
+    describe('authenticated', () => {
+      beforeAll(async () => {
+        authenticatedUser = await chattingUser.toJson()
+      })
+
+      describe('user id does not exist', () => {
+        it('returns null', async () => {
```

Comment:
i do not like that this returns `null` when its an error - but I assume thats because the query fails and returns null due to missing matching user.
```
MATCH (currentUser:User { id: $currentUserId })
MATCH (user:User { id: $userId })
```

One could consider to `if(!room) throw...`